ElectronClient/gui/NoteText.jsx

```diff
@@ -1461,7 +1461,12 @@ class NoteTextComponent extends React.Component {
 					let start = selectedStrings[i].search(/[^\s]/);
 					let end = selectedStrings[i].search(/[^\s](?=[\s]*$)/);
 					newBody += selectedStrings[i].substr(0, start) + string1 + selectedStrings[i].substr(start, end - start + 1) + string2 + selectedStrings[i].substr(end + 1);
-					if (this.state.note.body.substr(selection.end) === '') newBody = newBody.trim();
+
+					// correcting selction offset
```

Comment:
selction => selection

Also please add more information about what the code does exactly.

Reply:
`if (this.state.note.body.substr(selection.end) === '') newBody = newBody.trim();`

This was also added in #2582 

**_This has no underling logic only for improving aesthetics, this was removed to take care of a corner case while implementing this current PR._**

so while adding the markdown rule, we basically break the notes into 3 components
(before selected text) - selected text - (after selected text)

so we are checking that if (after selected text) is null remove all the white spaces after applying the closing the markdown rule.


#### With trim functionality
![trim-II](https://user-images.githubusercontent.com/54576074/76683923-c3a2a880-662d-11ea-920d-d50376a26741.gif)


#### Without trim functionality
![trim-I](https://user-images.githubusercontent.com/54576074/76683921-c1404e80-662d-11ea-82ab-781a445cbbb3.gif)